src/v/datalake/schema_registry.cc

```diff
@@ -57,40 +69,48 @@ get_proto_offsets_result get_schema_proto_offsets(iobuf& buf) {
 }
 
 result<proto_offsets_message_data, get_schema_error>
-get_proto_offsets(iobuf& buf) {
+get_proto_offsets(iobuf& buf) noexcept {
     proto_offsets_message_data result;
     iobuf_const_parser parser(buf.share(0, buf.size_bytes()));
 
     // The encoding is a length, followed by indexes into the file or message.
     // Each number is a zigzag encoded integer.
-    auto [offset_count, bytes_read] = parser.read_varlong();
-    if (!bytes_read) {
-        return get_schema_error::bad_varint;
-    }
-    // Reject more offsets than bytes remaining; it's not possible
-    if (static_cast<size_t>(offset_count) > parser.bytes_left()) {
-        return get_schema_error::not_enough_bytes;
-    }
-    if (offset_count == 0) {
-        result.protobuf_offsets.push_back(0);
-        result.shared_message_data = buf.share(
-          parser.bytes_consumed(), parser.bytes_left());
-        return result;
-    }
-    result.protobuf_offsets.resize(offset_count);
-    for (auto& o : result.protobuf_offsets) {
-        if (parser.bytes_left() == 0) {
-            return get_schema_error::not_enough_bytes;
-        }
-        std::tie(o, bytes_read) = parser.read_varlong();
+    try {
+        auto [offset_count, bytes_read] = parser.read_varlong();
         if (!bytes_read) {
             return get_schema_error::bad_varint;
         }
-    }
+        // Reject more offsets than bytes remaining; it's not possible
+        if (static_cast<size_t>(offset_count) > parser.bytes_left()) {
+            return get_schema_error::not_enough_bytes;
+        }
+        if (offset_count == 0) {
+            result.protobuf_offsets.push_back(0);
+            result.shared_message_data = buf.share(
+              parser.bytes_consumed(), parser.bytes_left());
+            return result;
+        }
+        result.protobuf_offsets.resize(offset_count);
+        for (auto& o : result.protobuf_offsets) {
+            if (parser.bytes_left() == 0) {
+                return get_schema_error::not_enough_bytes;
+            }
+            std::tie(o, bytes_read) = parser.read_varlong();
+            if (!bytes_read) {
+                return get_schema_error::bad_varint;
```

Comment:
does this not risk throwing on a short read?

Reply:
Chatter on slack, yeah this whole block is wrapped in a try/catch